### PR TITLE
Add hook for osgeo [<2.0] libraries

### PR DESCRIPTION
**Pull request by giumas**
This commit is based on: https://github.com/pyinstaller/pyinstaller/issues/1522

### CODE REVIEW

File: PyInstaller/loader/rthooks/pyi_rth_osgeo.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#-----------------------------------------------------------------------------
+
+import os
+import sys
+
+# Installing `osgeo` Conda packages requires to set `GDAL_DATA`
+
+is_win = sys.platform.startswith('win')
+if is_win:
+    gdal_data = os.path.join(sys._MEIPASS, 'Library', 'data')
```
Comment (matysek) [inline PyInstaller/loader/rthooks/pyi_rth_osgeo.py]: Would this hook work without rthook (without setting `GDAL_DATA` when you make the gdal_data path `sys._MEIPASS/osgeo/Library/data? Could you try that?

Comment (giumas) [inline PyInstaller/loader/rthooks/pyi_rth_osgeo.py]: It does not work without rthook. If the rthook is not called, the library cannot find the data.

**Comment by matysek** [comment]
Thanks.

File: PyInstaller/hooks/hook-osgeo.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#
+# TODO: Fix for gdal>=2.0: 'NameError: global name 'help' is not defined'
+
+# flag used to identify an Anaconda environment
+is_conda = False
+
+# Auxiliary data:
+#
+# - general case (data in 'osgeo/data/gdal'):
+datas = collect_data_files('osgeo', subdir='data/gdal')
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-osgeo.py]: Please use `os.path.join`. Using `/` as directory separator is not platform-independent.

File: PyInstaller/loader/rthooks/pyi_rth_osgeo.py (hunk 1 above)
Comment (htgoebel) [inline PyInstaller/loader/rthooks/pyi_rth_osgeo.py]: Why is an additional directory level used here?
Why not using a more expressive name for the directory, Like `osgeo`

Comment (giumas) [inline PyInstaller/loader/rthooks/pyi_rth_osgeo.py]: I agree that `Library/data` is not enough expressive, but I followed this solution to maintain the same folder structure as created by the `osgeo` Conda package. 

An use case for my solution is when the application sets the `GDAL_DATA` within the code (it is not so uncommon), and thus, when frozen, would point to `Library/data`, and not to the existing ''osgeo/data".

**Comment by htgoebel** [comment]
@giumas I'm curios why the commit of this pull-request is by "gmasetti", but the pull-request is by you. Since we are about to change the license of run-time hooks (see #4810), we need to clarify who is the author of this code.

**Comment by giumas** [comment]
Sorry for the delayed answer! Both users are me.. let me know if I need to take any action to fix it.

**Comment by Legorooj** [comment]
@giumas you don't need to take any action, there's no real problem. What we want to know, is would you be happy for the runtime hooks - any authored in PyInstaller - to be converted to Apache License 2.0? We need to change licenses just for those runtime hooks.

**Comment by giumas** [comment]
Yes, I am fine with the idea to adopt the Apache License.

**Comment by Legorooj** [comment]
@guimas thanks!

**Pull request merged**